packages/types/src/package.ts

```diff
@@ -72,16 +85,7 @@ interface BaseManifest {
   },
   cpu?: string[],
   os?: string[],
-  main?: string,
-  module?: string,
-  typings?: string,
-  types?: string,
-  publishConfig?: {
-    main?: string,
-    module?: string,
-    typings?: string,
-    types?: string,
-  },
+  publishConfig?: PublishConfigFields,
```

Comment:
I don't think there's value from listing all these fields here. Let's just make it a `publishConfig: Record<string, unknown>`

Reply:
Should the keys still be a whitelist or maybe a blacklist of fields that should not get copied (like `tag`, `registry` and `access`)?

A blacklist would allow to overwrite any field, eg remove build tool configurations (like `jest`, `eslintConfig`, ...) but could dangerously change the manifest.

a whitelist:

```js
const PUBLISH_CONFIG_WHITELIST = [
  // manifest fields that may make sense to overwrite
  'bin',
  'scripts',
  'config',
  'devDependencies',
  // unsure about these
  'optionalDependencies',
  'peerDependencies',
  'peerDependenciesMeta',
  'bundleDependencies',
  'bundledDependencies',
  // https://github.com/stereobooster/package.json#package-bundlers
  'main',
  'module',
  'typings',
  'types',
  'exports',
  'browser',
  'esnext',
  'es2015',
  'unpkg',
  'umd:main',
]
```

or blacklist:

```js
const PUBLISH_CONFIG_BLACKLIST = [
  // essential fields
  'name',
  'version',
  // unsure about these
  'optionalDependencies',
  'peerDependencies',
  'peerDependenciesMeta',
  'bundleDependencies',
  'bundledDependencies',
  // publish related
  'tag',
  'registry',
  'access',
  'allow-same-version',
]
```

The blacklist could/should probably include some more field from [npm-config](https://docs.npmjs.com/misc/config).

Reply:
oh, I completely forgot that publishConfig is used for [actual configuration as well](https://docs.npmjs.com/files/package.json#publishconfig). Now I am not sure it makes sense to extend this further.

If this is a valuable feature, we can come up with a design that has no limitations. Like a dedicated new field (for instance, `"publishConfig": { "overrides": {} }`). Or some convention for naming fields, like `"main.published"`

Reply:
cc @pnpm/collaborators

Reply:
I like both ideas (overrides object and `.published` suffix).

For our case it would be sufficient to copy the fields in the [updated whitelist](https://github.com/pnpm/pnpm/commit/d8a84fbce48bb16bb641b9f4f9f7a92c1ff0f2c9#diff-5bc9a713e4ba1c35d8ba8df6e35e12a1R65). These are the fields that may differ if a bundler is used or that may be different between development and production. In my opinion these are the most likely fields someone would like to override.

`scripts` and/or `config` may be out of scope for this PR and could be solved later by one of your proposed ideas.

I'm happy to adjust the PR.

Reply:
OK, let's do it without scripts and config for now.

Reply:
Done.

If this is ready to merge, I will update pnpm/pnpm.github.io#56 to include the new fields.

Reply:
it is fine but the tests are failing. The tests should be updated as well